ktest/httprecorder/request_log.go

```diff
@@ -0,0 +1,239 @@
+package httprecorder
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+type LogEntry struct {
+	Request  Request  `json:"request,omitempty"`
+	Response Response `json:"response,omitempty"`
+	Error    string   `json:"error,omitempty"`
+}
+
+type Request struct {
+	Method string      `json:"method,omitempty"`
+	URL    string      `json:"url,omitempty"`
+	Header http.Header `json:"header,omitempty"`
+	Body   string      `json:"body,omitempty"`
+}
+
+type Response struct {
+	Status     string      `json:"status,omitempty"`
+	StatusCode int         `json:"statusCode,omitempty"`
+	Header     http.Header `json:"header,omitempty"`
+	Body       string      `json:"body,omitempty"`
+}
+
+func (e *LogEntry) FormatHTTP() string {
+	var b strings.Builder
+	b.WriteString(e.Request.FormatHTTP())
+	b.WriteString(e.Response.FormatHTTP())
+	return b.String()
+}
+
+func (r *Request) FormatHTTP() string {
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf("%s %s\n", r.Method, r.URL))
+	var keys []string
+	for k := range r.Header {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, k := range keys {
+		for _, v := range r.Header[k] {
+			b.WriteString(fmt.Sprintf("%s: %s\n", k, v))
+		}
+	}
+	b.WriteString("\n")
+	if r.Body != "" {
+		b.WriteString(r.Body)
+		b.WriteString("\n\n")
+	}
+	return b.String()
+}
+
+func (l *RequestLog) ReplaceTimestamp() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+
+	for _, entry := range l.entries {
+		entry.Request.Body = resetTimestamp(entry.Request.Body)
+		entry.Response.Body = resetTimestamp(entry.Response.Body)
+	}
+}
+
+func resetTimestamp(body string) string {
```

Comment:
This method replaces `LastTransitionTime` on conditions, but does not care about other timestamps in the payload (e.g. `metadata.creationTimestamp`). Should it?